### ISSUE TITLE
elements with role="heading" don't appear in outline

### ISSUE THREAD

**Issue by jasonkiss** [validator/validator#506]
I'd expect that elements with `role="heading"` and optionally `aria-level` should appear as headings in the validator's document outline. For example, given the following HTML, the Outline should include a level one heading, followed by a level two heading and a level 3 heading:

```
<h1>A first level heading</h1>
<div role="heading">An ARIA "heading" with default level 2</div>
<div role="heading" aria-level="3">An ARIA "heading" at level 3</div>
```

**Comment by Ian-Y** [comment]
I'm seeing the same issue. The validator is being unable to recognize `<div role="heading" aria-level="2"></div>` and is showing warning message like: `Warning: Article lacks heading. Consider using h2-h6 elements to add identifying headings to all articles.`

**Comment by sideshowbarker** [comment]
See discussion at https://github.com/validator/validator/issues/688

The gist of it is, the outline feature is behaving as designed and there’s no plan to make any change to recognize `role=heading` and `aria-level` for the purposes of the outline.

But if you have new information, please feel free to comment here further and we could reopen this if needed.
